tests/distributions/test_multivariate.py

```diff
@@ -456,6 +456,33 @@ def test_sample_reparameterized(self):
         self.assertEqual(t_grads, None)
         self.assertEqual(logits_grads, None)
 
+    def test_path_derivative(self):
+        temperature = tf.ones([])
+        logits = tf.ones([2, 3])
+        n_samples = tf.placeholder(tf.int32, shape=[])
+
+        con_rep = ExpConcrete(temperature, logits, use_path_derivative=True)
+        samples = con_rep.sample(n_samples)
+        log_prob = con_rep.log_prob(samples)
+        t_path_grads, logits_path_grads = tf.gradients(log_prob, [temperature, logits])
+        sample_grads = tf.gradients(log_prob, samples)
+        t_true_grads = tf.gradients(samples, temperature, sample_grads)[0]
+        logits_true_grads = tf.gradients(samples, logits, sample_grads)[0]
```

Comment:
Nice use of `tf.gradients`. Didn't notice it supports initial gradients before.

Reply:
Yes, in fact `tf.gradients` is equivalent to `theano.tensor.Lop`.